galaxy_ng/app/api/ui/serializers/synclist.py

```diff
@@ -0,0 +1,106 @@
+import logging
+
+from rest_framework import serializers
+
+from pulp_ansible.app import models as pulp_ansible_models
+
+from galaxy_ng.app import models
+
+log = logging.getLogger(__name__)
+
+
+class SyncListCollectionSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = pulp_ansible_models.Collection
+        fields = ["namespace", "name"]
+
+
+class SyncListCollectionSummarySerializer(serializers.Serializer):
+    namespace = serializers.CharField(max_length=64)
+    name = serializers.CharField(max_length=64)
+
+
+class SyncListSerializer(serializers.ModelSerializer):
+    namespaces = serializers.SlugRelatedField(
+        many=True, slug_field="name", queryset=models.Namespace.objects.all()
+    )
+
+    # collections = SyncListCollectionSerializer(many=True)
+    collections = SyncListCollectionSummarySerializer(many=True)
+
+    groups = serializers.SlugRelatedField(
+        many=True, slug_field="name", queryset=models.Group.objects.all()
+    )
+
+    users = serializers.SlugRelatedField(
+        many=True, slug_field="username", queryset=models.User.objects.all()
+    )
+
+    def create(self, validated_data):
+        collections_data = validated_data.pop("collections")
+
+        namespaces_data = validated_data.pop("namespaces")
+
+        users_data = validated_data.pop("users")
+        groups_data = validated_data.pop("groups")
+
+        instance = models.SyncList.objects.create(**validated_data)
+        for collection_data in collections_data:
+            instance.collections.add(pulp_ansible_models.Collection.objects.get(**collection_data))
+
+        for namespace_data in namespaces_data:
+            instance.namespaces.add(namespace_data)
```

Comment:
This will use bulk insert:
```suggestion
        instance.namespaces.add(*namespace_data)
```